ldapAdapterEXT-ext-impl/src/main/java/com/bosch/osmi/sw360/ldapAdapter/CustomPortalLDAPImporterImpl.java

```diff
@@ -131,5 +164,80 @@ private Organization addOrganization(String organizationName, User user) throws
 		}
         return organization;
 	}
+
+	private String mapOrganizationName(String name) {
+		loadLDAPImporterProperties();
+		String defaultTarget = defaultToNull ? null : name;
+
+		if (matchPrefix) {
+			return organizationMappings.keySet().stream()
+					.filter(name::startsWith)
+					.sorted(Comparator.comparing(String::length).reversed())
```

Comment:
For performance reasons it might be good to sort the keySet once and then use
```
return sortedMappingKeys.stream()
    .filter(name::startsWith)
    .findFirst()
    [...]
```

Since this is lazy this continues once the first matching prefix is found.